bucoffea/monojet/monojetProcessor.py

```diff
@@ -381,24 +381,22 @@ def process(self, df):
                     region_weights.add('trigger', np.ones(df.size))
 
             if not df['is_data']:
-                if df['has_v_jet']:
-                    if re.match(r'.*_loose_v.*', region):
-                        region_weights.add('wtag_loose', evaluator['wtag_loose'](ak8.pt.max()))
-                    if re.match(r'.*_loosemd_v.*', region):
-                        region_weights.add('wtag_loosemd', evaluator['wtag_loosemd'](ak8.pt.max()))
-                    if re.match(r'.*_tight_v.*', region):
-                        region_weights.add('wtag_tight', evaluator['wtag_tight'](ak8.pt.max()))
-                    if re.match(r'.*_tightmd_v.*', region):
-                        region_weights.add('wtag_tightmd', evaluator['wtag_tightmd'](ak8.pt.max()))
-                else:
-                    if re.match(r'.*_loose_v.*', region):
-                        region_weights.add('wtag_mistag_loose', evaluator['wtag_mistag_loose'](ak8.pt.max()))
-                    if re.match(r'.*_loosemd_v.*', region):
-                        region_weights.add('wtag_mistag_loosemd', evaluator['wtag_mistag_loosemd'](ak8.pt.max()))
-                    if re.match(r'.*_tight_v.*', region):
-                        region_weights.add('wtag_mistag_tight', evaluator['wtag_mistag_tight'](ak8.pt.max()))
-                    if re.match(r'.*_tightmd_v.*', region):
-                        region_weights.add('wtag_mistag_tightmd', evaluator['wtag_mistag_tightmd'](ak8.pt.max()))
+                genVs = gen[(gen.pdg==23) + (gen.pdg==24) + (gen.pdg==-24) + (gen.pt>10)]
+                ak8_matched_mask = ak8.match(genVs, deltaRCut=0.8).flatten()
+                for wp in ['loose','loosemd','tight','tightmd']:
+                    if re.match(r'.*_{wp}_v.*', region):
+
+                        real_V_weights = evaluator[f'wtag_{wp}'](ak8.pt.max())
```

Comment:
can't we do
```python
matched_weights = evaluator[f'wtag_{wp}'](ak8[matched_mask]).prod() \
				  * evaluator[f'wtag_mistag_{wp}'](ak8[~matched_mask]).prod() 

```

Reply:
that would apply the correct SF for each jet, so if there are N correctly tagged jets, the weight is ~SF^N.

Since our analysis only looks at the *leading* ak8 jet, we should probably only look at the leading jet:

```python
matched_ak8 = ak8[matched_mask]
unmatched_ak8 = ak8[~matched_mask]
matched_weights = evaluator[f'wtag_{wp}'](matched_ak8[matched_ak8.pt.argmax()]).prod() \
				  * evaluator[f'wtag_mistag_{wp}'](unmatched_ak8[unmatched_ak8.pt.argmax()]).prod() 
```